src/vcpkg/base/files.cpp

```diff
@@ -39,6 +39,42 @@ namespace
         bool is_dot() const { return (last - first) == 1 && *first == '.'; }
         bool is_dot_dot() const { return (last - first) == 2 && *first == '.' && *(first + 1) == '.'; }
     };
+
+    void append_filesystem_stringish(std::string&) { }
+
+    template<class StringIsh>
+    void append_filesystem_stringish(std::string& target, StringIsh&& str)
+    {
+        vcpkg::StringView as_string_view{str};
+        target.push_back('"');
+        target.append(as_string_view.data(), as_string_view.size());
+        target.push_back('"');
+    }
+
+    template<class StringIsh0, class... StringIshN>
+    void append_filesystem_stringish(std::string& target, StringIsh0&& str0, StringIshN&&... strN)
```

Comment:
I could see an argument to just provide 3-4 overloads taking StringViews rather than this recursive templatey nonsense.

Reply:
I really don't like this recursive templatey nonsense, tbh. Also, it should at least take `const StringIsh&`, since you should _not_ be forwarding.

Reply:
Agreed that the 3-4 overloads is probably the best for this internal function.

However, I'd actually consider taking `std::initializer_list<StringView>` / `View<StringView>` as the argument to `exit_filesystem_call_error()` and just having the one function.